staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1092,6 +1093,27 @@ func FieldSelectorQueryParam(version string) string {
 	return "fieldSelector"
 }
 
+//MarshalJSON implements the json.Marshaler interface.
+func (meta ObjectMeta) MarshalJSON() ([]byte, error) {
```

Comment:
This will break encoding of all objects that inline objectmeta (which is all API objects) since it hijacks the MarshalJSON implementation of the containing object

Reply:
Indeed, the MarshalJSON function is promoted to the embedder. We will have to find another solution instead.

Reply:
It's my blunder. I didn't think thoroughly. As for Marshaller, I can only come up with specific Marshaller for each resource type, which is kind of cumbersome. Is there any suggestion?

Reply:
not really... I spent a while trying to figure out a solution in https://github.com/kubernetes/kubernetes/issues/67610 and couldn't come up with anything reasonable

Reply:
Maybe a bit too drastic, but what about creating a json fork / wrapper library that implements https://github.com/golang/go/issues/11939?